spec/models/affiliate_spec.rb

```diff
@@ -224,6 +224,109 @@
       end
     end
 
+    describe 'links json' do
+      subject(:affiliate) { described_class.create!(valid_create_attributes.merge(links_attributes)) }
+
+      context 'when links_json is valid' do
+        before do
+          affiliate.save!
+        end
+
+        context 'when no links are provided' do
+          let(:links_attributes) { {} }
```

Comment:
In general is better to use `Hash.new` rather than `{}` to make the code a little more readable.